### PR TITLE
Add report generator and test, move aggregate params

### PR DESCRIPTION
**Pull request by jspacek**
## Description
- This PR introduces the `ReportGenerator` which constructs the explain plan for issue #10.
- It also moves `AggregateParams` and associated classes to a separate file for ease of use. 

## Affected Dependencies
`DPEngine` will be responsible for adding the stages into the report.

## How has this been tested?
- `report_generator_test` uses a stubbed `aggregate` function to print out a report
- Tested with `conda` environment under `Python 3.8`

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: tests/report_generator_test.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+from pipeline_dp.aggregate_params import AggregateParams, Metrics
+from pipeline_dp.report_generator import ReportGenerator
+
+class ReportGeneratorTest(unittest.TestCase):
+  def test_report_empty(self):
+    self.assertEqual("", ReportGenerator(None).report())
+
+  def test_report_params(self):
+    test_report = ("Differential private: Computing metrics: "
```
Comment (dvadym) [inline tests/report_generator_test.py:13]: I like that text :)

File: pipeline_dp/aggregate_params.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+"""AggregateParameters containing parameters to pass to the DPEngine."""
```
Comment (dvadym) [inline pipeline_dp/aggregate_params.py:1]: maybe something like

Contains classes used for specifying DP aggregations parameters.

File: pipeline_dp/report_generator.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
+"""Generator for explain DP computations reports."""
+from pipeline_dp.aggregate_params import AggregateParams
+
+class ReportGenerator:
+  """Generate a report based on the metrics and stages in the pipeline.
```
Comment (dvadym) [inline pipeline_dp/report_generator.py:5]: Nit (i.e. style improvement): Generates

**Review by dvadym — CHANGES_REQUESTED** [review]
Thanks! It looks good. I left some minor comments

**Comment by jspacek** [comment]
> Thanks! It looks good. I left some minor comments

@dvadym Thank you for the review! Updated and ready for re-review 🎉

**Review by dvadym — APPROVED** [review]
Thank you!

**Comment by jspacek** [comment]
Thanks for the merge!

**Pull request merged**